### PR TITLE
Troubleshooting for NGINX error log: Connection Refused

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation ticket: https://mattermost.atlassian.net/browse/MM-37000

Updated:
- Updated shared installation topic Configuring NGINX with SSL and HTTP-2 to clarify that examples in the docs may not match network configuration, and to recommend using IP address over ``localhost`` in situations where the local machine resolves ``localhost`` to more than one IP address

### CODE REVIEW

File: source/install/config-proxy-nginx.rst [hunk 1]
```diff
@@ -21,10 +21,11 @@ On RHEL 7 and 8, open the file ``/etc/nginx/conf.d/mattermost``.
...
 
 .. note::
-  If you're going to use Let's Encrypt to manage your SSL certificate stop at step 3 here and please see the `NGINX HTTP/2 & SSL full configuration guide <https://docs.mattermost.com/install/config-ssl-http2-nginx.html>`__.
-
-.. note::
-  You'll need valid SSL certificates in order for NGINX to pin the certificates properly. Additionally, your browser must have permissions to accept the certificate as a valid CA-signed certificate.
+  
+  - If you're going to use Let's Encrypt to manage your SSL certificate, stop here at step 3 here and see the `NGINX HTTP/2 & SSL product documentation <https://docs.mattermost.com/install/config-ssl-http2-nginx.html>`__ for details.
+  - You'll need valid SSL certificates in order for NGINX to pin the certificates properly. Additionally, your browser must have permissions to accept the certificate as a valid CA-signed certificate.
+  - Examples in this documentation are using a private IP address, which may not match your network configuration.
```
Comment (justinegeffen) [inline source/install/config-proxy-nginx.rst:27]:
```suggestion
  - Note that the IP address included in the examples in this documentation may not match your network configuration.
```

File: source/install/config-proxy-nginx.rst [hunk 2]
```diff
@@ -214,13 +217,14 @@ For other troubleshooting tips for WebSocket errors, see `potential solutions he
       - "80"
       - "443"
 
-**Why does NGINX fail when installing Gitlab CE with Mattermost on Azure?**
+Why does NGINX fail when installing Gitlab CE with Mattermost on Azure?
```
Comment (justinegeffen) [inline source/install/config-proxy-nginx.rst:220]:
```suggestion
Why does NGINX fail when installing GitLab CE with Mattermost on Azure?
```

**Review by justinegeffen — COMMENTED** [review]
One rewording (which may or may not work, I can't decide) and one typo - otherwise looks good to me!

**Review by justinegeffen — APPROVED** [review]

File: source/install/config-ssl-http2-nginx.rst [hunk 3]
```diff
@@ -16,15 +16,18 @@ You can use any certificate that you want, but these instructions show you how t
...
+2. Open the your Mattermost ``nginx.conf`` file as *root* in a text editor, then update the ``{ip}`` address in the ``upstream backend`` to point towards Mattermost (such as ``127.0.0.1:8065``), and update the ``server_name`` to be your domain for Mattermost.
 
 .. note::
-   On Ubuntu this file is located at ``/etc/nginx/sites-available/``. If you don't have this file run ``sudo touch /etc/nginx/sites-available/mattermost``.
-   On CentOS/RHEL this file is located at ``/etc/nginx/conf.d/``. If you don't have this file run ``sudo touch /etc/nginx/conf.d/mattermost``.
+   
+   - On Ubuntu this file is located at ``/etc/nginx/sites-available/``. If you don't have this file, run ``sudo touch /etc/nginx/sites-available/mattermost``.
+   - On CentOS/RHEL this file is located at ``/etc/nginx/conf.d/``. If you don't have this file, run ``sudo touch /etc/nginx/conf.d/mattermost``.
+   - Examples in this documentation are using a private IP address, which may not match your network configuration.
```
Comment (cwarnermm) [inline source/install/config-ssl-http2-nginx.rst:29]:
```suggestion
   - Note that the IP address included in the examples in this documentation may not match your network configuration. 
```

**Review by coltoneshaw — CHANGES_REQUESTED** [review]
It looks good and works fine! My only thought is should we be slightly more explicit on when you'd use it? Ex:

"If you're running NGINX on the same machine as mattermost and it resolves ``localhost``"

**Comment by cwarnermm** [comment]
@coltoneshaw - Thank you so much! This additional context is a great addition!

**Review by coltoneshaw — APPROVED** [review]

**Pull request merged**